From 3073b9c4dec5e5877363794bf81cbd4b84fdb9ee Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Tue, 28 May 2024 16:36:41 +0200
Subject: [PATCH] Ensure that libzstd targets are promoted to global if they
 were found

Promote all internal zstd targets if they were found by WrapZSTD to
global using PROVIDED_TARGETS mechanism.

Amends 7d9d1220f367d9275dfaa7ce12e89b0a9f4c1978

Task-numer: QTBUG-119469
Change-Id: I15ec484304f7bf2b3ee2a533d2badb3bb7797863
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---
 configure.cmake | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: qt6-base/configure.cmake
===================================================================
--- qt6-base.orig/configure.cmake
+++ qt6-base/configure.cmake
@@ -109,7 +109,15 @@ SSL_free(SSL_new(0));
 }
 ")
 
-qt_find_package(WrapZSTD 1.3 PROVIDED_TARGETS WrapZSTD::WrapZSTD MODULE_NAME global QMAKE_LIB zstd)
+qt_find_package(WrapZSTD 1.3
+    PROVIDED_TARGETS
+        WrapZSTD::WrapZSTD
+        zstd::libzstd
+        zstd::libzstd_static
+        zstd::libzstd_shared
+    MODULE_NAME global
+    QMAKE_LIB zstd
+)
 qt_find_package(WrapDBus1 1.2 PROVIDED_TARGETS dbus-1 MODULE_NAME global QMAKE_LIB dbus)
 qt_find_package(Libudev PROVIDED_TARGETS PkgConfig::Libudev MODULE_NAME global QMAKE_LIB libudev)
 qt_find_package(LTTngUST PROVIDED_TARGETS LTTng::UST MODULE_NAME core QMAKE_LIB lttng-ust)
